src/Ocelot/LoadBalancer/LoadBalancers/LeastConnection.cs

```diff
@@ -61,7 +58,8 @@ public void Release(ServiceHostAndPort hostAndPort)
             lock (SyncLock)
             {
                 var matchingLease = _leases.FirstOrDefault(l => l.HostAndPort.DownstreamHost == hostAndPort.DownstreamHost
-                    && l.HostAndPort.DownstreamPort == hostAndPort.DownstreamPort);
+                    && l.HostAndPort.DownstreamPort == hostAndPort.DownstreamPort
+                    && l.HostAndPort.Scheme == hostAndPort.Scheme);
```

Comment:
OK I see the idea... additional comparison of `ServiceHostAndPort.Scheme`
We've already fixed it in PRs: #2111 and #2151